app/models/visit_group.rb

```diff
@@ -20,7 +20,8 @@ class VisitGroup < ActiveRecord::Base
   validates :arm_id,
             :name,
             presence: true
-  validates :day, presence: true, numericality: true
+  validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"
```

Comment:
Correct me if I am mistaken, there seem to be a syntax error in

unless: ENV.fetch('USE_EPIC'){false}

http://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless

I either need to add a new method name use_epic?  Use the :if option with the symbol :use_epic? or use the code I had earlier:
> +  validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"

I prefer to use the unless option with a string, the default can be set either to nil or false,  and it still work for those with EPIC turned on.


From: Andrew Cates [mailto:notifications@github.com]
Sent: Monday, September 21, 2015 2:54 PM
To: sparc-request/sparc-fulfillment
Cc: Ying Wu
Subject: Re: [sparc-fulfillment] added epic option, made visit_group.day validations conditional (#66)


In app/models/visit_group.rb<https://github.com/sparc-request/sparc-fulfillment/pull/66#discussion_r40017329>:

> @@ -20,7 +20,8 @@ class VisitGroup < ActiveRecord::Base

>    validates :arm_id,

>              :name,

>              presence: true

> -  validates :day, presence: true, numericality: true

> +  validates :day, presence: true, unless: "ENV.fetch('USE_EPIC'){nil} == 'false'"

Wouldn't it make more sense to use:

unless: ENV.fetch('USE_EPIC'){false}

That would default to false and still work for those with USE_EPIC set to 'true'

—
Reply to this email directly or view it on GitHub<https://github.com/sparc-request/sparc-fulfillment/pull/66/files#r40017329>.